jerry-core/ecma/builtin-objects/ecma-builtin-regexp-prototype.c

```diff
@@ -627,76 +481,143 @@ inline bool JERRY_ATTR_ALWAYS_INLINE
+    }
+    {
+      ecma_object_t *obj_p = ecma_get_object_from_value (this_arg);
+      ecma_extended_object_t *re_obj_p = (ecma_extended_object_t *) obj_p;
+
+      return ecma_builtin_regexp_prototype_compile (obj_p, re_obj_p, arguments_list_p[0], arguments_list_p[1]);
+    }
+  }
+#endif /* ENABLED (JERRY_BUILTIN_ANNEXB) */
+
+  /* 2. */
+  ecma_object_t *obj_p = ecma_get_object_from_value (this_arg);
+
+  switch (builtin_routine_id)
+  {
+    case ECMA_REGEXP_PROTOTYPE_ROUTINE_TEST:
+    {
+      return ecma_builtin_regexp_prototype_test (this_arg, arguments_list_p[0]);
+    }
+    case ECMA_REGEXP_PROTOTYPE_ROUTINE_EXEC:
+    {
+      return ecma_builtin_regexp_prototype_exec (this_arg, arguments_list_p[0]);
+    }
+    case ECMA_REGEXP_PROTOTYPE_ROUTINE_TO_STRING:
+    {
+      return ecma_builtin_regexp_prototype_to_string (obj_p);
+    }
+#if ENABLED (JERRY_ESNEXT)
+    case ECMA_REGEXP_PROTOTYPE_ROUTINE_SYMBOL_SEARCH:
+    {
+      return ecma_regexp_search_helper (this_arg, arguments_list_p[0]);
+    }
+    case ECMA_REGEXP_PROTOTYPE_ROUTINE_SYMBOL_MATCH:
+    {
+      return ecma_regexp_match_helper (this_arg, arguments_list_p[0]);
+    }
+    case ECMA_REGEXP_PROTOTYPE_ROUTINE_SYMBOL_REPLACE:
+    {
+      return ecma_regexp_replace_helper (this_arg, arguments_list_p[0], arguments_list_p[1]);
+    }
+    case ECMA_REGEXP_PROTOTYPE_ROUTINE_SYMBOL_SPLIT:
+    {
+      return ecma_regexp_split_helper (this_arg, arguments_list_p[0], arguments_list_p[1]);
+    }
+    /* The routines that do use obj_p */
+    case ECMA_REGEXP_PROTOTYPE_ROUTINE_GET_FLAGS:
+    {
+      return ecma_builtin_regexp_prototype_get_flags (obj_p);
+    }
+    case ECMA_REGEXP_PROTOTYPE_ROUTINE_GET_SOURCE:
+    {
+      ecma_extended_object_t *re_obj_p = (ecma_extended_object_t *) obj_p;
+
+      return ecma_builtin_regexp_prototype_get_source (re_obj_p);
+    }
+    case ECMA_REGEXP_PROTOTYPE_ROUTINE_GET_GLOBAL:
+    case ECMA_REGEXP_PROTOTYPE_ROUTINE_GET_IGNORE_CASE:
+    case ECMA_REGEXP_PROTOTYPE_ROUTINE_GET_MULTILINE:
+    case ECMA_REGEXP_PROTOTYPE_ROUTINE_GET_STICKY:
+    case ECMA_REGEXP_PROTOTYPE_ROUTINE_GET_UNICODE:
+    {
+      if (!ecma_object_is_regexp_object (this_arg))
+      {
+        return ecma_raise_type_error (ECMA_ERR_MSG ("'this' is not a RegExp object"));
+      }
+
+      uint16_t flags = RE_FLAG_EMPTY;
+      ecma_extended_object_t *re_obj_p = (ecma_extended_object_t *) obj_p;
+
+      ecma_value_t ret_value = ecma_builtin_regexp_prototype_flags_helper (re_obj_p, &flags);
+      if (ECMA_IS_VALUE_ERROR (ret_value))
+      {
+        return ret_value;
+      }
+
+      static const uint16_t re_flags[] =
+      {
+        RE_FLAG_GLOBAL, RE_FLAG_IGNORE_CASE,
+        RE_FLAG_MULTILINE, RE_FLAG_STICKY, RE_FLAG_UNICODE
+      };
+      uint16_t padded_index = (uint16_t) (builtin_routine_id - ECMA_REGEXP_PROTOTYPE_ROUTINE_GET_GLOBAL);
+      return ecma_make_boolean_value (flags & re_flags[padded_index]);
```

Comment:
I'd move this flag -> boolean login into `ecma_builtin_regexp_prototype_flags_helper`.